From a8c7a7a93e19b328a1a6e8114f21b2bff9ad4f69 Mon Sep 17 00:00:00 2001
From: Patrick Ohly <patrick.ohly@intel.com>
Date: Thu, 17 Nov 2016 18:00:46 +0100
Subject: [PATCH] fix --enable-xattr

Commit fc0f570d added a check that prevents using --enable-xattr
together with --enable-bsdtar, perhaps because it was assumed that this
wouldn't work because there is no special tar option as in the GNU tar
case.

But that combination works fine for (and is needed by) Ostro OS with
IMA and Smack xattrs, so the check needs to be removed.

Besides that, enabling xattrs also had no effect because xattrs.c
never got to see the SWUPD_WITH_XATTRS define due to not including
config.h.

Signed-off-by: Patrick Ohly <patrick.ohly@intel.com>
---
 configure.ac | 5 +----
 src/xattrs.c | 1 +
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 5cbc450..2166fd0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -135,10 +135,7 @@ AC_ARG_ENABLE(
 	AS_HELP_STRING([--enable-xattr],[Use extended file attributes (unused by default)])
 )
 AS_IF([test "x$enable_xattr" = "xyes"],
-	    [AC_DEFINE(SWUPD_WITH_XATTRS,1,[Use extended file attributes])
-	    AS_IF(test "x$enable_bsdtar" = "xyes",
-	    echo "Options --enable-bsdtar and --enable-xattr are incompatible" >&2
-	    AS_EXIT(1))],
+	    [AC_DEFINE(SWUPD_WITH_XATTRS,1,[Use extended file attributes])],
 	    [XATTR=no]
 )
 TARSELINUX="yes"
diff --git a/src/xattrs.c b/src/xattrs.c
index 36d4241..bbca237 100644
--- a/src/xattrs.c
+++ b/src/xattrs.c
@@ -27,6 +27,7 @@
 #include <string.h>
 #include <sys/xattr.h>
 
+#include "config.h"
 #include "swupd.h"
 #include "xattrs.h"
 
-- 
2.1.4

